### PR TITLE
Sowmika, Bindu | MOBN-938 | Display service failure error messages

### PR DESCRIPTION
**Pull request by binduak**

### CODE REVIEW

File: ui/react-components/components/AddAppointment/AddAppointment.jsx [hunk 1]
```diff
@@ -222,45 +222,68 @@ const AddAppointment = props => {
         setShowSuccessPopup(true);
     };
 
+    const setServiceErrorMessageFromResponse = response => {
+        response.error && response.error.message ? setServiceErrorMessage(response.error.message)
```
Comment (AlekhyaYalla) [inline ui/react-components/components/AddAppointment/AddAppointment.jsx]: does it need response null check?
response && response.error && ...?

File: ui/react-components/components/EditAppointment/EditAppointment.jsx [hunk 2]
```diff
@@ -203,14 +204,24 @@ const EditAppointment = props => {
 
     const isWeeklyRecurringAppointment = () => appointmentDetails.recurrenceType === weekRecurrenceType;
 
+    const setServiceErrorMessageFromResponse = response => {
+        response.error && response.error.message ? setServiceErrorMessage(response.error.message)
```
Comment (AlekhyaYalla) [inline ui/react-components/components/EditAppointment/EditAppointment.jsx]: Same as in add appointment comment.

Comment (AlekhyaYalla) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: Can we extract to helper file, we use it in both edit and add.?

File: ui/react-components/components/AddAppointment/AddAppointment.jsx (hunk 1 above)
Comment (binduak) [reply ui/react-components/components/AddAppointment/AddAppointment.jsx]: @AlekhyaYalla We are doing this check where ever we are calling the **setServiceErrorMessageFromResponse** . so its not required here

File: ui/react-components/components/EditAppointment/EditAppointment.jsx (hunk 2 above)
Comment (binduak) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: same applies as the above

File: ui/react-components/components/AddAppointment/AddAppointment.jsx (hunk 1 above)
Comment (AlekhyaYalla) [reply ui/react-components/components/AddAppointment/AddAppointment.jsx]: okay

File: ui/react-components/components/EditAppointment/EditAppointment.jsx (hunk 2 above)
Comment (binduak) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: extracting to helper is a good idea.. I will see that

Comment (binduak) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: I tried but we are depending on errorTranslations.unexpectedServiceErrorMessage it has dependency on **ErrorTranslationsUtil**

**Pull request merged**